Makefile.am

```diff
@@ -1,52 +1,57 @@
-DESTDIR ?= /
-prefix ?= $(DESTDIR)
-bindir=/usr/bin
-datadir ?= /usr/share
-sysconfdir ?= /etc
-systemd_unitdir ?= /lib/systemd
-docexamplesdir ?= /usr/share/doc/mender-client/examples
-
-VERSION = $(shell git describe --tags --dirty --exact-match 2>/dev/null || git rev-parse --short HEAD)
-
-IDENTITY_SCRIPTS = \
-	support/mender-device-identity
-
-INVENTORY_SCRIPTS = \
-	support/mender-inventory-bootloader-integration \
-	support/mender-inventory-hostinfo \
-	support/mender-inventory-network \
-	support/mender-inventory-os \
-	support/mender-inventory-provides \
-	support/mender-inventory-rootfs-type \
-	support/mender-inventory-update-modules
-
-INVENTORY_NETWORK_SCRIPTS = \
-	support/mender-inventory-geo
-
-MODULES = \
-	support/modules/deb \
-	support/modules/docker \
-	support/modules/directory \
-	support/modules/single-file \
-	support/modules/rpm \
-	support/modules/script
-
-MODULES_ARTIFACT_GENERATORS = \
-	support/modules-artifact-gen/docker-artifact-gen \
-	support/modules-artifact-gen/directory-artifact-gen \
-	support/modules-artifact-gen/single-file-artifact-gen
-
-DBUS_POLICY_FILES = \
-	support/dbus/io.mender.AuthenticationManager.conf \
-	support/dbus/io.mender.UpdateManager.conf
-
-build: mender
+ACLOCAL_AMFLAGS = -I m4
+
+SUBDIRS = common vendor mender-auth mender-update
+
+check-local: test
+
+test: main_test
+	./main_test $(TESTFLAGS)
+.PHONY: test
+
+coverage: Makefile
+	rm -rf reports/*.xml
+	$(MAKE) \
+		CXXFLAGS="$(CXXFLAGS) --coverage" \
+		TESTFLAGS="$(TESTFLAGS) --gtest_output=xml:reports/" \
+		test
+	lcov --capture \
+		--quiet \
+		--directory . \
+		--output-file coverage.lcov \
+		--exclude '/usr/*' \
+		--exclude '*/googletest/*' \
+		--exclude '*_test.*'
+.PHONY: coverage
+
+vendor/googletest/lib/libgtest.a:
+	( cd vendor/googletest && cmake . && make )
+
+main_test: main_test.cpp lib.cpp Makefile vendor/googletest/lib/libgtest.a
+	$(CXX) \
+		-o main_test \
+		lib.cpp \
+		main_test.cpp \
+		-pthread \
+		vendor/googletest/lib/libgtest.a \
+		vendor/googletest/lib/libgtest_main.a \
+		-I vendor/googletest/googletest/include \
+		$(CXXFLAGS) \
+		$(LDFLAGS)
+
+# ============= the rest of this file is to be removed =============
+
+all-local: mender
+
+clean-local:
+	rm -f mender main_test
+	( cd vendor/googletest && git clean -df )
 
 mender: main.cpp lib.cpp
-	g++ -o mender main.cpp lib.cpp $(CXXFLAGS) $(LDFLAGS)
+	$(CXX) -o mender main.cpp lib.cpp $(CXXFLAGS) $(LDFLAGS)
```

Comment:
Q: What about `CPPFLAGS`?

Reply:
I'm guessing this is going away anyhow - same as the files it references?

Reply:
Yes, but we might as well have it right in the git history. :+1: